omnigibson/utils/usd_utils.py

```diff
@@ -228,6 +228,139 @@ def create_joint(prim_path, joint_type, body0=None, body1=None, enabled=True,
+        """
+        return cls._PATH_TO_IDX[prim_path]
+
+    @classmethod
+    def get_all_impulses(cls):
+        """
+        Grab all impulses at the current timestep
+
+        Returns:
+            n-array: (N, N, 3) impulse array defining current impulses between all N contact-sensor enabled rigid bodies
+                in the simulator
+        """
+        # Generate the contact matrix if it doesn't already exist
+        if cls._CONTACT_MATRIX is None:
+            cls._CONTACT_MATRIX = cls._CONTACT_VIEW.get_contact_force_matrix(dt=1.0)
+            cls._CONTACT_CACHE = dict()
+
+        return cls._CONTACT_MATRIX
+
+    @classmethod
+    def get_impulses(cls, prim_paths_a, prim_paths_b):
+        """
+        Grabs the matrix representing all impulse forces between rigid prims from @prim_paths_a and
+        rigid prims from @prim_paths_b
+
+        Args:
+            prim_paths_a (list of str): Rigid body prim path(s) with which to grab contact impulses against
+                any of the rigid body prim path(s) defined by @prim_paths_b
+            prim_paths_b (list of str): Rigid body prim path(s) with which to grab contact impulses against
+                any of the rigid body prim path(s) defined by @prim_paths_a
+
+        Returns:
+            n-array: (N, M, 3) impulse array defining current impulses between N bodies from @prim_paths_a and M bodies
+                from @prim_paths_b
+        """
+        # Compute subset of matrix and return
+        idxs_a = [cls._PATH_TO_IDX[path] for path in prim_paths_a]
+        idxs_b = [cls._PATH_TO_IDX[path] for path in prim_paths_b]
+        return cls.get_all_impulses()[idxs_a][:, idxs_b]
+
+    @classmethod
+    def in_contact(cls, prim_paths_a, prim_paths_b):
+        """
+        Check if any rigid prim from @prim_paths_a is in contact with any rigid prim from @prim_paths_b
+
+        Args:
+            prim_paths_a (list of str): Rigid body prim path(s) with which to check contact against any of the rigid
+                body prim path(s) defined by @prim_paths_b
+            prim_paths_b (list of str): Rigid body prim path(s) with which to check contact against any of the rigid
+                body prim path(s) defined by @prim_paths_a
+
+        Returns:
+            bool: Whether any body from @prim_paths_a is in contact with any body from @prim_paths_b
+        """
+        # Check if the contact tuple already exists in the cache; if so, return the value
+        key = (tuple(prim_paths_a), tuple(prim_paths_b))
+        in_contact = None if cls._CONTACT_CACHE is None else cls._CONTACT_CACHE.get(key, None)
+        if in_contact is not None:
+            return in_contact
+
+        # In contact if any of the matrix values representing the interaction between the two groups is non-zero
+        in_contact = np.any(cls.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
+
+        cls._CONTACT_CACHE[key] = in_contact
+
+        return in_contact
```

Comment:
nit, a cleaner way to write this could be:
```
key = (tuple(prim_paths_a), tuple(prim_paths_b))
if key not in cls._CONTACT_CACHE:
    cls._CONTACT_CACHE[key]  = np.any(cls.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
return cls._CONTACT_CACHE[key]
```

Reply:
done